sqlmesh/core/environment.py

```diff
@@ -102,32 +103,46 @@ class Environment(EnvironmentNamingInfo):
         previous_finalized_snapshots: Snapshots that were part of this environment last time it was finalized.
     """
 
-    snapshots: t.List[SnapshotTableInfo]
+    snapshots_: t.List[t.Any] = Field(alias="snapshots")
```

Comment:
why are you using t.Any? that bound is too wide

Reply:
Workaround discussed with @izeigerman for Pydantic model limitations regarding unions. The type checking is now done in the validator.